### PR TITLE
Moving towards a custom drive class rather than RLU's

### PR DESCRIPTION
**Pull request by robbai**

### CODE REVIEW

**Comment by robbai** [comment]
Also has a half-flip which needs a bit of tuning, takes place of "self.dodge" in the agent (Hypebot).

**Comment by jeroen11dijk** [comment]
Why does the halfflip replace the dodge?

**Comment by robbai** [comment]
> Why does the halfflip replace the dodge?

Would you like halfflip to be a separate state?

File: bot/derevo.py [hunk 1]
```diff
@@ -162,13 +167,17 @@ def get_controls(self):
                 self.dodge.target = self.their_goal.center
         elif self.step is Step.Defending:
             defending(self)
-        elif self.step is Step.Dodge:
-            self.dodge.step(self.fps)
-            if self.dodge.finished and self.info.my_car.on_ground:
+        elif self.step is Step.Dodge or self.step is Step.HalfFlip:
```
Comment (jeroen11dijk) [inline bot/derevo.py:170]: why not make a seperate statement for halfflipping?

**Review by jeroen11dijk — APPROVED** [review]

**Pull request merged**